frontend/src/util/Hook.tsx

```diff
@@ -30,6 +30,52 @@ export const useBestSubmission = (player?: Player | null) => {
   return bestSubmission;
 };
 
+export const useGetScore = (player?: Player) => {
+  const counted = new Set<number>();
+  const [score, setScore] = useState<number>(0);
+
+  useEffect(() => {
+    if (player) {
+      for (let i = 0; i < player.submissions.length; i += 1) {
+        if (player.submissions[i].numCorrect === player.submissions[i].numTestCases
+          && !counted.has(player.submissions[i].problemIndex)) {
+          counted.add(player.submissions[i].problemIndex);
+        }
+      }
+
+      setScore(counted.size);
+    }
+  }, [player, setScore, counted]);
```

Comment:
Just to double check, do sets count as "state" in some form? As in just making sure that the counted variable isn't like cleared/reset on every re-render (or something similar that could cause bugs) given that it's not part of core React

Reply:
I'm not so sure what that means, but if set is cleared, the next time that the player object changes it will be refilled. We can talk about this in the call.